bindata/assets/alerts/client-go-requests.yaml

```
@@ -0,0 +1,53 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: client-go
+  namespace: openshift-kube-apiserver
+spec:
+  groups:
+    - name: control-plane-client-go
+      rules:
+        # latency by:
+        # - job (kubelet, apiserver. controller-manager, ...)
+        # - destination host: internal-lb, kubernetes.default service, direct connection ...
+        - record: cluster:rest_client_request_duration_seconds:internal_load_balancer:histogram_quantile
+          labels:
+            quantile: "0.99"
+          expr: |
+            histogram_quantile(0.99,
+              sum(rate(
+                label_replace(rest_client_request_duration_seconds_bucket{verb="GET",url=~"https?://api-int.*"},
+                  "host","$1","url","https?://([^/\\s]+).*")[5m:30s]
+                )
+              ) by (le,host,service,namespace,node)
+            )
+        - record: cluster:rest_client_request_duration_seconds:services:histogram_quantile
+          labels:
+            quantile: "0.99"
+          expr: |
+            histogram_quantile(0.99,
+              sum(rate(
+                label_replace(rest_client_request_duration_seconds_bucket{verb="GET",url!~"https?://(api-int|\\[::1\\]|127\\.0\\.0\\.1|localhost).*"},
+                  "host","$1","url","https?://([^/\\s]+).*")[5m:30s]
+                ) * on (cluster_ip) group_left group(kube_service_info{cluster_ip!~"None",namespace=~"(openshift|kube|default).*"}) by (service,namespace,cluster_ip)
```

This part seems superflux:
```suggestion
                ) * on (cluster_ip) group_left kube_service_info{cluster_ip!~"None",namespace=~"(openshift|kube|default).*"}
```

---

we need to aggregate or it can fail with errors like
```
found duplicate series for the match group {cluster_ip="None"} on the right hand-side of the operation: [{__name__="kube_service_info", cluster_ip="None", container="kube-rbac-proxy-main", endpoint="https-main", job="kube-state-metrics", namespace="openshift-cluster-machine-approver", prometheus="openshift-monitoring/k8s", service="machine-approver"}, {__name__="kube_service_info", cluster_ip="None", container="kube-rbac-proxy-main", endpoint="https-main", job="kube-state-metrics", namespace="kube-system", prometheus="openshift-monitoring/k8s", service="kubelet"}];many-to-many matching not allowed: matching labels must be unique on one side
```

---

it seems that services with external_ips are duplicated too, we have to aggregate by service/namespace that is unique, but I don't know if that is the correct statement though, I just assumed group was doing that and it actually works

---

there are some services that are referred by url
I think that we better use here the rest of connections that doesn't use the load balancer or the localhost address